app/controllers/groups_controller.rb

```diff
@@ -0,0 +1,24 @@
+class GroupsController < ApplicationController
+  skip_before_action :authenticate_user!
+
+  def index
+    @groups = Group.all
```

Comment:
let's have `Group.where` preference: preference (it's a bit more complicated than that, try to look into it and if you can't figure it out let me know)
the preference comes from params - if there are no preferences in params let's use the preferences that the current_user has selected